CRM/Extension/Manager.php

```diff
@@ -280,6 +280,8 @@ public function install($keys) {
     $this->statuses = NULL;
     $this->mapper->refresh();
     CRM_Core_Invoke::rebuildMenuAndCaches(TRUE);
+
+    \Civi::reset();
```

Comment:
@totten any concern with this line?

Reply:
I *think* the `\Civi::reset()` isn't needed if `CRM_Logging_Schema::fixSchemaDifferencesFor()` is called with the second parameter (as used here):
https://github.com/civicrm/civicrm-core/blob/9634dbac5efa59168f493f101f8691c1f448b0a2/CRM/Core/BAO/CustomField.php#L1717

With that parameter, it hits a different code path where it doesn't used the cached info and adjust based on the field you pass in.

The `CRM_Logging_Schema::fixSchemaDifferencesFor` and `Civi::service('sql_triggers')->rebuild()` bit might be worth extracting as it would then be almost duplicated both in `createField` and `bulkSave`.

Reply:
@eileenmcnaughton ,

@pfigel 's suggestion to pass the second parameter with the list of added columns worked perfectly. I've removed the reset from `CRM_Extension_Manager::install()`.